test/JDBC/expected/1_grant_schema.out

```diff
@@ -0,0 +1,2936 @@
+-- tsql
+drop login l1;
+go
+
+-- tsql
+-- Create objects for tests in a user defined schema
+create schema s1;
+go
+create login l1 with password = '123';
+go
+create user u1 for login l1;
+go
+create login l2 with password = '123';
+go
+create user u2 for login l2;
+go
+create table s1.t1(a int);
+go
+create view s1.v1 as select 1;
+go
+create procedure s1.p1 as select 1;
+go
+create function s1.f1() returns int begin declare @a int; set @a = 1; return @a; end
+go
+create trigger s1.tr1 ON s1.t1 for insert as begin 
+select 'insert successful' end
+go
+create trigger s1.tr2 ON s1.t1 for insert as begin 
+grant select on s1.t1 to u1 end
+go
+create sequence s1.sq1 start with 1 increment by 1 ;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 2 [users have no default permission]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql
+grant select on schema::s1 to u1;
+go
+grant insert on schema::s1 to u1;
+go
+grant delete on schema::s1 to u1;
+go
+grant update on schema::s1 to u1;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 3 [user u1 has select, insert, delete privilege on the schema]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
+go
+~~START~~
+int
+~~END~~
+
+insert into s1.t1 values (1);
+go
+~~START~~
+varchar
+insert successful
+~~END~~
+
+~~ROW COUNT: 1~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for sequence sq1)~~
```

Comment:
Few questions:

1. This is expected right? I assume so because u1 does not have the necessary permission for sq1.
2. Do we have any way to grant permission of sq1 to u1 in BBF? Is it there in T-SQL? If so we should document if not done already.
3. In T-SQL does GRANT \<some permission\> ON SCHEMA, GRANT on sequence as well? If so and if we are unable to support this at the moment in BBF we should also document this if not already done.

Reply:
1. yes, this is expected.
2. Yes, we do have update privilege on sequences which we don't support right now in BBF.
3. [BABEL-4821] will support it later.